digits_test.go

```diff
@@ -6,251 +6,341 @@ import (
+			fractionalPrecision: digits.PreserveUpToHundredth,
+		},
+		output: Output{
+			sigFigs:    "338863",
+			nonSigFigs: "000",
+			strOut:     "(338,863,000.00)",
+			head:       "(",
+			core:       "338,863",
+			tail:       ",000.00)",
+		},
 	},
 	{
-		p: digits.Thousands,
-		v: "45121000",
-		g: ',',
-		d: digits.PreserveUpToHundredth,
+		stimulus: Stimulus{
+			precision:           digits.Exact,
+			value:               "1038807",
+			groupSeparator:      ',',
+			fractionalPrecision: digits.PreserveUpToHundredth,
+		},
+		output: Output{
+			sigFigs:    "1038807",
+			nonSigFigs: "",
+			strOut:     "1,038,807.00",
+			head:       "",
+			core:       "1,038,807.",
+			tail:       "00",
+		},
 	},
 	{
-		p: digits.Thousands,
-		v: "98000",
-		g: ',',
-		d: digits.PreserveUpToHundredth,
+		stimulus: Stimulus{
+			precision:           digits.HundredThousands,
+			value:               "80800000",
+			groupSeparator:      ',',
+			fractionalPrecision: digits.PreserveUpToHundredth,
+		},
+		output: Output{
+			sigFigs:    "808",
+			nonSigFigs: "00000",
+			strOut:     "80,800,000.00",
+			head:       "",
+			core:       "80,8",
+			tail:       "00,000.00",
+		},
 	},
 	{
-		p: digits.Thousands,
-		v: "-338863000",
-		g: ',',
-		d: digits.PreserveUpToHundredth,
+		stimulus: Stimulus{
+			precision:           digits.HundredThousands,
+			value:               "-80800000",
+			groupSeparator:      ',',
+			fractionalPrecision: digits.PreserveUpToHundredth,
+		},
+		output: Output{
+			sigFigs:    "808",
+			nonSigFigs: "00000",
+			strOut:     "(80,800,000.00)",
+			head:       "(",
+			core:       "80,8",
+			tail:       "00,000.00)",
+		},
 	},
 	{
-		p: digits.Exact,
-		v: "1038807",
-		g: ',',
-		d: digits.PreserveUpToHundredth,
+		stimulus: Stimulus{
+			precision:           digits.Tenth,
+			value:               "80800000.99090909090",
+			groupSeparator:      ',',
+			fractionalPrecision: digits.PreserveUpToHundredth,
+		},
+		output: Output{
+			sigFigs:    "80800000.9",
+			nonSigFigs: "9",
+			strOut:     "80,800,000.99",
+			head:       "",
+			core:       "80,800,000.9",
+			tail:       "9",
+		},
 	},
 }
 
-func act() []*digits.Expression {
-	var suts = []*digits.Expression{}
-	for _, s := range stimuli {
-		d, err := digits.New(s.p, s.v, s.g, s.d)
-		if err != nil {
-			panic(err)
-		}
-		suts = append(suts, d)
+func process(stimulus Stimulus) *digits.Expression {
+	d, err := digits.New(stimulus.precision, stimulus.value, stimulus.groupSeparator, stimulus.fractionalPrecision)
+	if err != nil {
+		panic(err)
 	}
-	return suts
+	return d
 }
 func Test_SigFigs(t *testing.T) {
-	var suts = act()
-	expectations := []string{
-		"0.",
-		"0.02",
-		"0.02",
-		"77",
-		"77",
-		"77",
-		"396006",
-		"487",
-		"4059",
-		"45121",
-		"98",
-		"338863",
-		"1038807",
-	}
-	for i, expectation := range expectations {
-		sut := suts[i]
-		output := sut.SigFigs()
-		if output != expectation {
+	for _, sut := range suts {
```

Comment:
Similar issue here. See my previous comments about cases, sut, and output.

Reply:
addressed in https://github.com/joshuanario/digits/commit/bccb9198881484d18b62ea3e63ae9dd5f7bda76b. testCase and testCases used instead of case and cases, due to case being a reserved name in go